napari/_vendor/qt_json_builder/qt_jsonschema_form/widgets.py

```diff
@@ -696,14 +712,32 @@ def populate_from_schema(
             else:
                 description = ""
 
+            if 'allOf' in sub_schema:
+                # replace reference in schema with the reference values.
+                # this will work for the values we have now, but may need
+                # a json reader that can parse the references in many cases 
+                # (see jsonref for example)
+                ref_name = sub_schema['allOf'][0]['$ref'].split('/')[2]
+                ref_values = schema['definitions'][ref_name]
+                sub_schema['allOf'][0] = ref_values
+                
+
             sub_ui_schema = ui_schema.get(name, {})
             widget = widget_builder.create_widget(
                 sub_schema, sub_ui_schema, description=description
             )  # TODO onchanged
             widget._name = name
             widget.on_changed.connect(partial(self.widget_on_changed, name))
             label = sub_schema.get("title", name)
-            layout.addRow(label, widget)
+
+            
+            if isinstance(widget, QtWidgets.QWidget):
+                layout.addRow(label, widget)
+            else: 
+                #this is probably a magic gui widget.
+                layout.addRow(label, widget._widget.native)
```

Comment:
again private api (our api) instead of proper method/property in `BaseMagicSetting`